zk.js/tests/verifiers.test.ts

```diff
@@ -246,42 +250,47 @@ describe("Verifier tests", () => {
       lightWasm,
       assets: [FEE_ASSET, MINT],
       amounts: [new BN(shieldFeeAmount), new BN(shieldAmount)],
-      publicKey: account.keypair.publicKey,
+      owner: account.keypair.publicKey,
     });
+
     inputUtxo = createOutUtxo({
       lightWasm,
       assets: [FEE_ASSET, MINT],
       amounts: [new BN(shieldFeeAmount), new BN(shieldAmount)],
-      publicKey: account.keypair.publicKey,
+      owner: account.keypair.publicKey,
       metaHash: BN_1,
       address: BN_2,
     });
+
     inputUtxo2 = createOutUtxo({
       lightWasm,
       assets: [FEE_ASSET],
       amounts: [BN_0],
-      publicKey: account.keypair.publicKey,
+      owner: account.keypair.publicKey,
       metaHash: new BN(3),
       address: new BN(4),
     });
-    inputProgramUtxo = createOutUtxo({
+
+    inputProgramUtxo = createProgramOutUtxo({
       lightWasm,
       assets: [FEE_ASSET],
       amounts: [BN_0],
-      publicKey: hashAndTruncateToCircuit(
-        getVerifierProgramId(IDL_PUBLIC_LIGHT_PSP2IN2OUT).toBytes(),
-      ),
+      owner: getVerifierProgramId(IDL_PUBLIC_LIGHT_PSP2IN2OUT),
       metaHash: new BN(6),
       address: new BN(7),
-      utxoDataHash: BN_1,
-      utxoData: { rnd: 1 },
+      data: { rnd: 1 },
+      ownerIdl: IDL_PUBLIC_LIGHT_PSP2IN2OUT,
+      /// Allows for custom hashing schema
+      dataHash: BN_1,
+      // dataHash: createDataHashWithDefaultHashingSchema({ rnd: 1 }, lightWasm),
```

Comment:
where does it get the dataHash from?

Reply:
shouldn't this throw an error?

Reply:
we're passing BN_1 as dataHash and {rnd:1} as data; then BN_1 represents a custom hashing in this case